user/yaml/README.md

```diff
@@ -0,0 +1,327 @@
+secret "kubestellar" deleted
+secret/kubestellar created
+Ready!
+```
+
+Note that alternatively, one can wait for the `kubestellar` secret to be created.
+
+```shell
+kubectl wait -n kubestellar \
+  --for=condition=ready pod \
+  --selector=app=kubestellar-server \
+  --timeout=120s
+
+until kubectl logs -n kubestellar $(kubectl get pod -n kubestellar --selector=app=kubestellar-server -o jsonpath={.items[0].metadata.name}) 2>&1 | grep -Fxq "Ready!"
+do
+  sleep 1
+done
+```
+
+After the deployment has completed, **KubeStellar** `admin.kubeconfig` can be in three ways:
+
+- the `kubestellar` secret in the `kubestellar` namespace or any other namespace listed in `SECRET_NAMESPACES` environment variable;
+- directly from the `kubestellar` pod in the `kubestellar` namespace at the location `/home/kubestellar/.kcp/external.kubeconfig`.
+
+## Deploy **KubeStellar** in an **OpenShift** cluster
+
+In this case use the following YAML to deploy  **KubeStellar** `stable` in a `kubestellar` namespace:
+
+```shell
+kubectl apply -f kubestellar-server-route.yaml
+```
+
+Then follow the same instructions for the Kind cluster deployment.
+
+## Access **KubeStellar** service directly from the host OS without `admin.kubeconfig` or plugins
+
+Since **kubectl**, **kcp** plugins, and **KubeStellar** executables are include in the **KubeStellar** container image we can operate KubeStellar directly from the host OS using `kubectl`, for example:
+
+```shell
+$ kubectl exec -n kubestellar $(kubectl get pod -n kubestellar --selector=app=kubestellar-server -o jsonpath={.items[0].metadata.name}) -- kubectl ws tree
+.
+└── root
+    ├── compute
+    └── espw
+
+$ kubectl exec -n kubestellar $(kubectl get pod -n kubestellar --selector=app=kubestellar-server -o jsonpath={.items[0].metadata.name}) -- kubectl ws create imw
+Workspace "imw" (type root:organization) created. Waiting for it to be ready...
+Workspace "imw" (type root:organization) is ready to use.
+```
+
+## Access **KubeStellar** service from the host OS by extracting the `admin.kubeconfig` and the plugins from the pod
+
+In this case the host OS will need a copy of **kcp** plugins, and **KubeStellar** plugins, assuming that the architecture of the client is the same of the container, the executables can be extracted from the container:
+
+```shell
+kubectl exec -n kubestellar $(kubectl get pod -n kubestellar --selector=app=kubestellar-server -o jsonpath={.items[0].metadata.name}) -- tar cf - "/home/kubestellar/kcp-plugins" | tar xf - --strip-components=2
+
+kubectl exec -n kubestellar $(kubectl get pod -n kubestellar --selector=app=kubestellar-server -o jsonpath={.items[0].metadata.name}) -- tar cf - "/home/kubestellar/kubestellar" | tar xf - --strip-components=2
+```
+
+As an alternative the bootstrap script can be used with the `--deploy false` option as follows:
+
+```shell
+bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/main/bootstrap/bootstrap-kubestellar.sh) --deploy false

```

Comment:
This also creates a version mismatch problem; these instructions are embedded in one particular version of KubeStellar, but (IIRC) this command will fetch a version determined by the website contents at the moment the command is executed.

Reply:
would be good to add `--kcp-version` and -`-kubestellar-version` flags with a default latest and ask the user to change.
Another option is to extract the running version from the container and set it.

Reply:
I have added the following:

```
In this use case, it may be necessary to specify the version of **KubeStaller** with the `--kubestellar-version` argument to match the version deployed in the cluster.
```

@ronenkat We do not like you to choose the version of kcp, it will be decided by kubestellar

Reply:
There are better solutions than asking users to be aware of whether we have changed the stable image in a relevant way and calling for an older version of the image.

The simplest approach is to have the Makefile reference the container image by digest rather than a release-numbered tag.  That way, making behavior changes to the container image in the future does not disrupt existing users.

The disadvantage of the simplest approach is that delivering any change, even a bug fix, to users requires making a new release or telling those users to use a branch or tag that is not a release.  That is in fact the full general story in any case, so all we are talking about here is a special case for this container image.  We _could_ support this special case using a command line option to specify the container image tag or digest; with this, a user could get a image behavior change without getting a new release or non-release version of KubeStellar by specifying the new container digest on the command line.  Or we could simply say that new behavior is delivered in a release or use of a non-released version of KubeStellar, period.

If we really want to prepare for delivering changes in container image _behavior_ that do not change the _interface_ (that is, the aspects of behavior that matter to usage from the outside) of the image then we could adopt a practice of tagging the image with a name for the version of image's interface (rather than a release number of the whole KubeStellar repo).

Reply:
@francostellari, I didn't recall logic in the bootstrap script that infer the kcp version from the kubestellar version

Reply:
@ronenkat it's because currently we only support v0.11.0, it may change in the future